### PR TITLE
[Merged by Bors] - feat: `Prod.fst : X × Y → X` is a closed map if `Y` is compact

### PR DESCRIPTION
**Pull request by urkud**
From the Mandelbrot set connectedness project

Co-Authored-By: @girving 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Constructions.lean [hunk 1]
```diff
@@ -491,6 +491,10 @@ theorem continuous_swap : Continuous (Prod.swap : α × β → β × α) :=
   continuous_snd.prod_mk continuous_fst
 #align continuous_swap continuous_swap
 
+lemma isClosedMap_swap : IsClosedMap (Prod.swap : α × β → β × α) := fun s hs ↦ by
+  rw [image_swap_eq_preimage_swap]
+  exact hs.preimage continuous_swap
```
Comment (alreadydone) [inline Mathlib/Topology/Constructions.lean:496]: I think you can inline a proof using [Homeomorph.prodComm](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Homeomorph.html#Homeomorph.prodComm) with [Homeomorph.isClosedMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Homeomorph.html#Homeomorph.isClosedMap) without needing adding this lemma.

Comment (urkud) [reply Mathlib/Topology/Constructions.lean:496]: This file doesn't know about `Homeomorph`s (yes, I think that we should reorder imports but I would prefer to do it later).

**Comment by girving** [comment]
Thank you!

File: Mathlib/Topology/Constructions.lean (hunk 1 above)
Comment (ADedecker) [reply Mathlib/Topology/Constructions.lean:496]: I agree that we should reorder files, but also that this is fine for now.

**Comment by ADedecker** [comment]
Thanks! Let me mention that this should be a result about proper maps, but since we already have the `fst` version there's nothing wrong about adding the `snd` version and potentially refactoring both later on.

bors r+

**Pull request closed without merging**